File: dry_run_tool.py
import re
from pathlib import Path
from check_file_permission import check_file_permission

def dry_run_replace(file_path, pattern, replacement, config, regex=False):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()
    new_content = re.sub(pattern, replacement, content) if regex else content.replace(pattern, replacement)
    return f"[Dry Run] Would update {file_path}:\n{new_content}"

File: ts-tools/dry_run_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function dryRunReplace(
  filePath: string,
  pattern: string,
  replacement: string,
  config: any,
  regex: boolean = false,
): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const newContent = regex
              ? data.replace(new RegExp(pattern, 'g'), replacement)
              : data.replace(pattern, replacement);
            resolve(`[Dry Run] Would update ${filePath}:\n${newContent}`);
          }
        });
      })
      .catch(reject);
  });
}
